historydag/dag.py

```diff
@@ -2227,6 +2227,51 @@ def accum_between_clade(dictlist):
             ).values()
         )[0]
 
+    def underestimate_rf_diameter(self):
+        """Returns an underestimate of the RF diameter of the DAG.
+        This estimate is calculated by calculating the maximal sum RF distance
+        between the DAG and a random tree from a topological outlier.
+
+        On a set of DAGs with 2000 or less histories, this underestimate
+        is quite accurate compared to the actual computed RF diameter.
+        """
+        dag_copy = self.copy()
+        dag_copy.trim_optimal_sum_rf_distance(dag_copy, optimal_func=max)
+        ref_history = dag_copy.sample()
+        return self.optimal_rf_distance(ref_history, optimal_func=max)
+
+    def overestimate_rf_diameter(self):
+        """Returns an overestimate of the RF diameter of the DAG.
+        This estimate is calculated by calculating twice of the maximal sum RF
+        distance between the DAG and a random tree from the median tree.
+
+        On a set of DAGs with 2000 or less histories, this underestimate was
+        not close compared to the actual RF diameter. However, the
+        overestimate was never more than twice of the actual RF diameter.
+        """
+        dag_copy = self.copy()
+        dag_copy.trim_optimal_sum_rf_distance(dag_copy, optimal_func=min)
+        ref_history = dag_copy.sample()
+        return 2 * self.optimal_rf_distance(ref_history, optimal_func=max)
+
+    def optimal_one_sided_sum_rf_distance(
```

Comment:
I think this should be called just `optimal_one_sided_rf_distance` since we're not taking a sum over anything... unless I'm misunderstanding!

Reply:
I think the method returns something like the sum of one-sided RF distances rather than a single distance. I tried running it one some examples, and it returns a number that is bigger than the one returned by `optimal_rf_distance`. 

Also, from skimming the code in `utils.one_side_rfdistance_funcs` it looks pretty close to `utils.sum_rfdistance_fucs`